app/Models/Wiki/Image.php

```diff
@@ -67,10 +68,23 @@ public function sync($force = false)
             return $this;
         }
 
-        $lock = cache()->lock($this->cacheKey().':lock', 300);
+        $cache = cache();
+        $lock = $cache->lock($this->cacheKey().':lock', 300);
 
         if (!$lock->get()) {
-            return $this;
+            if ($this->cache !== null) {
+                return $this;
+            }
+
+            try {
+                $lock->block(10);
+                $lock->release();
```

Comment:
should this be in a `finally`?

Reply:
I don't think so because in other cases the lock isn't obtained

Reply:
🤔 
(assuming the follow lines don't explode)

Reply:
actually this can't even be in finally as otherwise the lock would still be held when calling sync for return